client/state/plugins/installed/selectors-ts.ts

```diff
@@ -95,3 +95,55 @@ export const getAllPluginsIndexedBySiteId = createSelector(
 	},
 	( state ) => [ getAllPluginsIndexedByPluginSlug( state ), getSiteIdsThatHavePlugins( state ) ]
 ) as { ( state: AppState ): { [ siteId: number ]: { [ pluginSlug: string ]: Plugin } } };
+
+export function getPluginsOnSites( state: AppState, plugins: Plugin[] ) {
+	return Object.values( plugins ).reduce(
+		( acc: { [ pluginSlug: string ]: Plugin }, plugin: Plugin ) => {
+			const siteIds = Object.keys( plugin.sites ).map( Number );
+			const pluginOnSites = getPluginOnSites( state, siteIds, plugin.slug );
+			if ( pluginOnSites ) {
+				acc[ plugin.slug ] = pluginOnSites;
+			}
+			return acc;
+		},
+		{}
+	);
+}
+
+export function getPluginOnSites( state: AppState, siteIds: number[], pluginSlug: string ) {
+	const plugin = getAllPluginsIndexedByPluginSlug( state )[ pluginSlug ];
+
+	if ( ! plugin ) {
+		return undefined;
+	}
+
+	for ( const siteId of siteIds ) {
+		if ( plugin.sites[ siteId ] ) {
+			return plugin;
+		}
+	}
+
+	return undefined;
+}
+
+export function getPluginOnSite( state: AppState, siteId: number, pluginSlug: string ) {
+	const plugin = getAllPluginsIndexedByPluginSlug( state )[ pluginSlug ];
+
+	const { sites, ...pluginWithoutSites } = plugin;
+
+	return plugin && plugin.sites[ siteId ]
```

Comment:
We could have an early return and avoid the ternary.

Reply:
Good call.